pkg/controller/clusterpool/collections.go

```diff
@@ -213,17 +213,57 @@ type cdCollection struct {
 	byClaimName map[string]*hivev1.ClusterDeployment
 }
 
-func isBroken(cd *hivev1.ClusterDeployment) bool {
+// NOTE: This doesn't care about claimed or deleted/deleting status. That's on the caller.
+func isBroken(cd *hivev1.ClusterDeployment, pool *hivev1.ClusterPool, logger log.FieldLogger) bool {
+	////
+	// Check for ProvisionStopped
+	////
 	cond := controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions, hivev1.ProvisionStoppedCondition)
 	if cond == nil {
 		// Since we should be initializing conditions, this probably means the CD is super fresh.
 		// Don't declare it broken yet -- give it a chance to come to life.
 		return false
 	}
 	if cond.Status == corev1.ConditionTrue {
+		logger.Infof("Cluster %s is broken due to ProvisionStopped", cd.Name)
 		return true
 	}
-	// Other causes of broken-ness to be added here later. E.g. resume failed.
+
+	////
+	// Check for resume timeout
+	////
+	if pool.Spec.HibernationConfig == nil || pool.Spec.HibernationConfig.ResumeTimeout.Duration == 0 {
+		// If no timeout was configured, skip this check.
+		return false
+	}
+	if !cd.Spec.Installed {
+		// Don't bother checking on a cluster that isn't installed yet.
+		return false
+	}
+	if cd.Spec.PowerState != hivev1.ClusterPowerStateRunning {
+		// If spec.powerState isn't Running, we're not resuming.
+		return false
+	}
+	if cd.Status.PowerState == hivev1.ClusterPowerStateHibernating || cd.Status.PowerState == hivev1.ClusterPowerStateRunning {
+		// If the hibernation controller hasn't yet kicked off resume; or if the cluster has already
+		// completed resuming, skip.
+		return false
+	}
+	cond = controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions, hivev1.ClusterHibernatingCondition)
+	if cond == nil {
+		// Since we should be initializing conditions, this probably means the CD is super fresh.
+		// Don't declare it broken yet -- give it a chance to come to life.
+		return false
+	}
+	if cond.Reason != hivev1.HibernatingReasonResumingOrRunning {
+		// It would be weird if we got here, but better safe.
```

Comment:
If you choose to remove the powerstate check above, then this comment needs to be changed